---
title: "Pretty_Deseq2_Elizabeth"
output: html_document
date: '2022-05-04'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

---
title: "Deseq2_tximport"
output: html_document
date: '2022-05-02'
---

# Analysis for RNAseq Elizabeth's Project 

## Logic/ Steps for Analysis 

-   Download from SRA using GEO

    -   GSE143897

    -   SRA meta data and accession list were also downloaded

    -   Done on Cheaha

-   Formatted with FASTQ using SRA toolkit

    -   Done on Cheaha

-   Used Salmon for alignment for count data

    -   Done on Cheaha

    -   Focus was for counts

    -   Did not need variants so I did not use STAR aligner

-   Tximport is a package to import count data from Salmon so that they can be imported into DESeq2

    -   Mike Love's (author for most of the well used RNAseq packages) guide uses either tximport or txmeta

    -   tximport was the suggestion from Lara Ianov when txmeta did not cooperate with the merge off Cheaha and onto local computer

    -   This was done with R and R Studio on my local Mac

-   DESeq2 is the standard publication wise for normalizing counts

    -   This was done with R and R Studio on my local Mac

    -   Standard for normalization and analysis in most publications

    -   Did the following groups

        -   Platinum vs Control

        -   Tumor vs Ascites vs Control

        -   Tumor vs Benign

        -   Tumor vs Ascites

    -   Includes graphs for each gene of interest for each comparison group


```{r message= FALSE, warning= FALSE, echo=FALSE}
library(tximport)
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(DESeq2)
```


```{r message= FALSE, warning= FALSE, echo=FALSE}
txdb <- makeTxDbFromGFF(file="~/Downloads/gencode.v19.annotation.gtf")
saveDb(x=txdb, file = "gencode.v19.annotation.gtf_withproteinids.TxDb")
k <- keys(txdb, keytype = "TXNAME")
tx2gene <- select(txdb, k, "GENEID", "TXNAME")
```

## Group 1: Tumor, Ascites, and Control

## Control vs Platinum 

```{r message= FALSE, warning= FALSE, echo=FALSE}
dir <- "~/Documents/Mythreye_Lab/quants"
samples <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/sample_info_tximport.csv")
files <- file.path(dir, samples$SRR, "quant.sf")
all(file.exists(files))
txi <- tximport(files, type = "salmon", tx2gene= tx2gene, ignoreAfterBar= TRUE)
names(txi)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
sample_meta_data <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/Elizabeth_Sample_meta.csv")
sampleFiles <- samples$SRR
sampleCondition <- sample_meta_data$Treatment
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r echo=FALSE}
colnames(txi$counts) <- rownames(sampleTable)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

### Graphs for Genes of Interest

-   Each circle is a sample in the respective group

-   X axis: group

-   Y axis: Counts

#### INHBA

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000122641.9", intgroup="condition")
```

#### ACVR2A

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000121989.10", intgroup="condition")
```

#### ACVR2B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000114739.9", intgroup="condition")
```

#### FOXP3

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000049768.10", intgroup="condition")
```

#### p53

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000141510.11", intgroup="condition")
```

#### BRCA1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000012048.15", intgroup="condition")
```

#### BRCA2

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000139618.10", intgroup="condition")
```

#### INHBB

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000163083.5", intgroup="condition")
```

#### ACVR1B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000135503.8", intgroup="condition")
```

#### ACVR1C

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000123612.11", intgroup="condition")
```

#### TGFB1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000105329.5", intgroup="condition")
```

#### CD8A
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000153563.11", intgroup="condition")
```

#### CD3G
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000160654.5", intgroup="condition")
```


#### SOX2
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000181449.2", intgroup="condition")
```


## Tumor vs Ascites vs Control

```{r message= FALSE, warning= FALSE, echo=FALSE}
sampleFiles <- samples$SRR
sampleCondition <- sample_meta_data$Tumor..Normal
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r echo=FALSE}
colnames(txi$counts) <- rownames(sampleTable)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r, echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

### Graphs for Genes of Interest

-   Each circle is a sample in the respective group

-   X axis: group

-   Y axis: Counts

#### INHBA

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000122641.9", intgroup="condition")
```

#### ACVR2A

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000121989.10", intgroup="condition")
```

#### ACVR2B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000114739.9", intgroup="condition")
```

#### FOXP3

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000049768.10", intgroup="condition")
```

#### p53

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000141510.11", intgroup="condition")
```

#### BRCA1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000012048.15", intgroup="condition")
```

#### BRCA2

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000139618.10", intgroup="condition")
```

#### INHBB

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000163083.5", intgroup="condition")
```

#### ACVR1B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000135503.8", intgroup="condition")
```

#### ACVR1C

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000123612.11", intgroup="condition")
```

#### TGFB1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000105329.5", intgroup="condition")
```

#### CD8A
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000153563.11", intgroup="condition")
```

#### CD3G
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000160654.5", intgroup="condition")
```

#### SOX2
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000181449.2", intgroup="condition")
```

## Group 2: Tumor vs Benign


```{r message= FALSE, warning= FALSE, echo=FALSE}
dir <- "~/Documents/Mythreye_Lab/quants"
samples <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/Benign_Tumor/data_import_benign_vs_tumor.csv")
files <- file.path(dir, samples$SRR, "quant.sf")
all(file.exists(files))
txi <- tximport(files, type = "salmon", tx2gene= tx2gene, ignoreAfterBar= TRUE)
names(txi)
```


```{r echo=FALSE}
sample_meta_data <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/Benign_Tumor/Elizabeth_meta_benign_vs_tumor.csv")
```

```{r echo=FALSE}
sampleFiles <- samples$SRR
sampleCondition <- sample_meta_data$Tumor..Normal
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r echo=FALSE}
colnames(txi$counts) <- rownames(sampleTable)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

### Graphs for Genes of Interest

-   Each circle is a sample in the respective group

-   X axis: group

-   Y axis: Counts

#### INHBA

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000122641.9", intgroup="condition")
```

#### ACVR2A

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000121989.10", intgroup="condition")
```

#### ACVR2B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000114739.9", intgroup="condition")
```

#### FOXP3

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000049768.10", intgroup="condition")
```

#### p53

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000141510.11", intgroup="condition")
```

#### BRCA1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000012048.15", intgroup="condition")
```

#### BRCA2

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000139618.10", intgroup="condition")
```

#### INHBB

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000163083.5", intgroup="condition")
```

#### ACVR1B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000135503.8", intgroup="condition")
```

#### ACVR1C

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000123612.11", intgroup="condition")
```

#### TGFB1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000105329.5", intgroup="condition")
```

#### CD8A
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000153563.11", intgroup="condition")
```

#### CD3G
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000160654.5", intgroup="condition")
```

#### SOX2
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000181449.2", intgroup="condition")
```


## Group 3: Ascites vs Benign 

```{r message= FALSE, warning= FALSE, echo=FALSE}
dir <- "~/Documents/Mythreye_Lab/quants"
samples <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/Benign_Tumor/data_import_ascites_vs_normal.csv")
files <- file.path(dir, samples$SRR, "quant.sf")
all(file.exists(files))
txi <- tximport(files, type = "salmon", tx2gene= tx2gene, ignoreAfterBar= TRUE)
names(txi)
```

```{r echo=FALSE}
sample_meta_data <- read.csv(file= "~/Documents/Mythreye_Lab/Elizabeth/MetaData/Benign_Tumor/Elizabeth_meta_ascites_vs_normal.csv")
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
sampleFiles <- samples$SRR
sampleCondition <- sample_meta_data$Tumor..Normal
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r echo=FALSE}
colnames(txi$counts) <- rownames(sampleTable)
```

### Part 2: DEseq2 for Ascites vs Benign 

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

### Graphs for Genes of Interest

-   Each circle is a sample in the respective group

-   X axis: group

-   Y axis: Counts

#### INHBA

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000122641.9", intgroup="condition")
```

#### ACVR2A

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000121989.10", intgroup="condition")
```

#### ACVR2B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000114739.9", intgroup="condition")
```

#### FOXP3

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000049768.10", intgroup="condition")
```

#### p53

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000141510.11", intgroup="condition")
```

#### BRCA1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000012048.15", intgroup="condition")
```

#### BRCA2

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000139618.10", intgroup="condition")
```

#### INHBB

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000163083.5", intgroup="condition")
```

#### ACVR1B

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000135503.8", intgroup="condition")
```

#### ACVR1C

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000123612.11", intgroup="condition")
```

#### TGFB1

```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000105329.5", intgroup="condition")
```

#### CD8A
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000153563.11", intgroup="condition")
```

#### CD3G
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000160654.5", intgroup="condition")
```

#### SOX2
```{r echo=FALSE}
plotCounts(dds, gene= "ENSG00000181449.2", intgroup="condition")
```
